### PR TITLE
Issue1913 remove sta t

### PR DESCRIPTION
**Pull request by hcasperfu**
This closes https://github.com/ibpsa/modelica-ibpsa/issues/1913.

The models that have `sta_?.T` in their reference results:

`Fluid.HeatExchangers.Radiators.Examples.RadiatorEN442_2`
`Fluid.FixedResistances.Validation.LosslessPipe`
`Examples.{SimpleHouse,Tutorial.SimpleHouse.SimpleHouse6}`
- Added `TemperatureTwoPort` components to replace `sta_?.T`.

`Fluid.Interfaces.Examples.{PrescribedOutlet,PrescribedOutlet_TSetOnly,PrescribedOutlet_Water,PrescribedOutlet_X_wSetOnly}`
- Added `TemperatureTwoPort` to base class `IBPSA.Fluid.Interfaces.Examples.BaseClasses.PrescribedOutletState`.

`Fluid.SolarCollectors.Validation.{EN12975_Array,EN12975_Series}`
`Fluid.Humidifiers.Validation.{SteamHumidifier_X_dynamic,SprayAirWasher_X,SteamHumidifier_X,SprayAirWasher_X_dynamic}`
`Fluid.Interfaces.Examples.{ConservationEquation,StaticTwoPortConservationEquation}`
~~`Fluid.Interfaces.Examples.Humidifier_u`~~
- No action. Flow rate is constant.

`Fluid.Geothermal.Borefields.Examples.RectangularBorefield`
- No action because `allowFlowReversal==false`.

### CODE REVIEW

**Comment by hcasperfu** [comment]
Converted PR to draft.
`Fluid.Interfaces.Examples.Humidifier_u` is in the list in https://github.com/OpenModelica/OpenModelica/issues/12704 and I need more time to think about what to do with it.

**Comment by hcasperfu** [comment]
Converted to draft again to work on the following additional models:
- `Fluid.Chillers.Validation.Carnot_TEva_reverseFlow`
- `Fluid.HeatExchangers.Validation.ConstantEffectiveness`
- `Fluid.HeatPumps.Validation.Carnot_TCon_reverseFlow`

They were missed in the `grep` because variable names with the regex form `sta_..\.T` (such as `sta_a1.T`) were not considered.

**Comment by mwetter** [comment]
@jelgerjansen : Can you please review this so that we have a non-LBNL reviewer. This change is needed due to allow verification across different Modelica tools. See https://github.com/lbl-srg/modelica-buildings/issues/3957 for the reason and discussions.

**Review by mwetter — APPROVED** [review]

File: IBPSA/Fluid/FixedResistances/Validation/LosslessPipe.mo [hunk 1]
```diff
@@ -26,22 +26,36 @@ model LosslessPipe "Validation model for lossless pipe"
...
 
   IBPSA.Fluid.FixedResistances.LosslessPipe res(
     redeclare package Medium = Medium,
     show_T=true,
     m_flow_nominal=1)
     "Fixed resistance"
-    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+    annotation (Placement(transformation(extent={{10,-10},{30,10}})));
 
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTemInl(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/FixedResistances/Validation/LosslessPipe.mo:38]: `senTemIn`

File: IBPSA/Fluid/HeatExchangers/Radiators/Examples/RadiatorEN442_2.mo [hunk 2]
```diff
@@ -56,50 +56,74 @@ model RadiatorEN442_2 "Test model for radiator"
...
     offset=300000 + dp_nominal,
     height=-dp_nominal)
-    annotation (Placement(transformation(extent={{-100,-60},{-80,-40}})));
+    annotation (Placement(transformation(extent={{-100,-20},{-80,0}})));
   Modelica.Thermal.HeatTransfer.Sources.FixedTemperature TBCRad2(T=TRoo)
-    annotation (Placement(transformation(extent={{-32,-20},{-20,-8}})));
+    annotation (Placement(transformation(extent={{-32,-24},{-20,-12}})));
   Modelica.Thermal.HeatTransfer.Sources.FixedTemperature TBCRad1(T=TRoo)
     annotation (Placement(transformation(extent={{-32,48},{-20,60}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTemRad1Inl(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/HeatExchangers/Radiators/Examples/RadiatorEN442_2.mo:69]: `senTemRad1In`

File: IBPSA/Fluid/HeatExchangers/Radiators/Examples/RadiatorEN442_2.mo [hunk 3]
```diff
@@ -56,50 +56,74 @@ model RadiatorEN442_2 "Test model for radiator"
...
+    redeclare final package Medium = Medium,
+    m_flow_nominal=rad1.m_flow_nominal,
+    tau=0) "Temperature sensor at inlet of radiator 1"
+    annotation (Placement(transformation(extent={{-40,-2},{-20,18}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTemRad1Out(
+    redeclare final package Medium = Medium,
+    m_flow_nominal=rad1.m_flow_nominal,
+    tau=0) "Temperature sensor at outlet of radiator 1"
+    annotation (Placement(transformation(extent={{20,-2},{40,18}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTemRad2Inl(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/HeatExchangers/Radiators/Examples/RadiatorEN442_2.mo:79]: `senTemRad2In`

File: IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo [hunk 4]
```diff
@@ -48,53 +48,75 @@ model Carnot_TCon_reverseFlow
...
     nPorts=1,
     redeclare package Medium = Medium2)
-    annotation (Placement(transformation(extent={{-10,-10},{10,10}}, origin={-50,-30})));
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}}, origin={-70,-30})));
   Modelica.Blocks.Math.Gain mEva_flow(k=-1/cp2_default/dTEva_nominal)
     "Evaporator mass flow rate"
     annotation (Placement(transformation(extent={{40,-90},{60,-70}})));
   Modelica.Blocks.Math.Add QEva_flow(k2=-1) "Evaporator heat flow rate"
     annotation (Placement(transformation(extent={{40,-50},{60,-30}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTConEnt(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo:71]: `senTConIn`

File: IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo [hunk 5]
```diff
@@ -48,53 +48,75 @@ model Carnot_TCon_reverseFlow
...
+    redeclare final package Medium = Medium1,
+    final m_flow_nominal=m1_flow_nominal,
+    tau=0) "Temperature sensor for fluid entering condenser"
+    annotation (Placement(transformation(extent={{-40,-4},{-20,16}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTConLvg(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo:76]: `senTConOut`

File: IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo [hunk 6]
```diff
@@ -48,53 +48,75 @@ model Carnot_TCon_reverseFlow
...
+    redeclare final package Medium = Medium1,
+    final m_flow_nominal=m1_flow_nominal,
+    tau=0) "Temperature sensor for fluid leaving condenser"
+    annotation (Placement(transformation(extent={{40,20},{60,40}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTEvaLvg(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo:81]: `senTEvaOut`

File: IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo [hunk 7]
```diff
@@ -48,53 +48,75 @@ model Carnot_TCon_reverseFlow
...
+    redeclare final package Medium = Medium2,
+    final m_flow_nominal=heaPum.m2_flow_nominal,
+    tau=0) "Temperature sensor for fluid leaving evaporator"
+    annotation (Placement(transformation(extent={{-20,-40},{-40,-20}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort senTEvaEnt(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/HeatPumps/Validation/Carnot_TCon_reverseFlow.mo:86]: `senTEvaIn`

File: IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo [hunk 8]
```diff
@@ -110,6 +110,46 @@ partial model PrescribedOutletState "Test model for prescribed outlet state"
     T=293.15,
     nPorts=1) "Flow source"
     annotation (Placement(transformation(extent={{-90,-100},{-70,-80}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort THeaHigPowInl(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo:113]: `THeaHigPowIn`

File: IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo [hunk 9]
```diff
@@ -110,6 +110,46 @@ partial model PrescribedOutletState "Test model for prescribed outlet state"
...
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final tau=0) "Temperature sensor at the inlet of heaHigPow"
+    annotation (Placement(transformation(extent={{-60,76},{-40,96}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort THeaHigPowOut(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final tau=0) "Temperature sensor at the outlet of heaHigPow"
+    annotation (Placement(transformation(extent={{20,76},{40,96}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort TCooLimPowInl(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo:123]: `TCooLimPowIn`

File: IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo [hunk 10]
```diff
@@ -110,6 +110,46 @@ partial model PrescribedOutletState "Test model for prescribed outlet state"
...
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final tau=0) "Temperature sensor at the inlet of cooLimPow"
+    annotation (Placement(transformation(extent={{-60,-10},{-40,10}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort TCooLimPowOut(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final tau=0) "Temperature sensor at the outlet of cooLimPow"
+    annotation (Placement(transformation(extent={{20,-10},{40,10}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort THeaCooUnlInl(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo:133]: `THeaCooUnlIn`

File: IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo [hunk 11]
```diff
@@ -110,6 +110,46 @@ partial model PrescribedOutletState "Test model for prescribed outlet state"
...
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final tau=0) "Temperature sensor at the inlet of heaCooUnl"
+    annotation (Placement(transformation(extent={{-60,-70},{-40,-50}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort THeaCooUnlOut(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final tau=0) "Temperature sensor at the outlet of heaCooUnl"
+    annotation (Placement(transformation(extent={{20,-70},{40,-50}})));
+  IBPSA.Fluid.Sensors.TemperatureTwoPort TSteStaInl(
```
Comment (jelgerjansen) [inline IBPSA/Fluid/Interfaces/Examples/BaseClasses/PrescribedOutletState.mo:143]: `TSeStaInl`

**Review by jelgerjansen — COMMENTED** [review]
The changes look good to me, I only have a minor comment regarding the naming of temperature sensors at the inlet of a component. If you want to keep the current naming, that's fine by me (and then I'll approve)

**Comment by hcasperfu** [comment]
@jelgerjansen -
Greetings and thanks for the review!
I changed the variable names as suggested except for the chiller/HP models. Personally I think "entering" and "leaving" temperatures are more commonly used than "in" or "out" in these contexts. An example in the library is `IBPSA.Fluid.Chillers.Examples.Carnot_TEva`, although it is not for a temperature sensor.

**Review by jelgerjansen — APPROVED** [review]

**Pull request merged**